plaidml/edsl/edsl.h

```diff
@@ -806,7 +806,13 @@ inline Tensor floor(const Tensor& x) { return intrinsic("floor", x); }
 /// \param y Tensor
 /// \return Tensor
 ///
-inline Tensor gather(const Tensor& x, const Tensor& y) { return intrinsic("gather", x, y); }
+
+inline Tensor gather(const Tensor& x, const Tensor& y, const std::vector<Tensor>& attributes = {}) {
+  std::vector<Tensor> args = {x};
+  args.emplace_back(y);
+  args.insert(args.end(), attributes.begin(), attributes.end());
+  return intrinsicCall("gather", args);
+}
```

Comment:
We should use a fluent/builder-style API if we want the user to specify optional attributes/values to the op. See how we did this for something like a convolution.

Reply:
Use a builder-style API for gather now with default attributes values.  Currently the same default values are used in 3 places, 1, edsl c++ API. 2, edsl python API. 3. tile GatherOp (in tile/ir/ops.td file).